src/schema/FroidSchema.ts

```diff
@@ -862,4 +875,15 @@ export class FroidSchema {
   private static parseSchema(schema: string): DocumentNode {
     return parse(schema, {noLocation: true});
   }
+
+  /**
+   * Checks if there are any shareable types in the FROID object types.
+   *
+   * @returns {boolean} True if there are shareable types, false otherwise.
+   */
+  private hasShareableTypes(): boolean {
+    return Object.values(this.froidObjectTypes).some(
+      (objectType) => objectType.isShareable
```

Comment:
Calling out that the `@shareable` directive can be applied to fields. So we should probably check for it there as well. We should also support only including the fields with `@shareable` when the directive is applied at the field level and not the object level.

https://www.apollographql.com/docs/graphos/schema-design/federated-schemas/reference/directives